src/tasks/command/command_parser.c

```diff
@@ -0,0 +1,56 @@
+/**
+ * @author  Thomas Haile
+ * @date    2025-05-24
+ *
+ * Command parsing implementation
+ */
+
+#include "command_parser.h"
+#include "macros.h"
+
+/// @brief Parse packet and dispatch command to appropriate queue
+void dispatch_command(slate_t *slate, packet_t *packet)
+{
+    slate->number_commands_processed++;
+
+    Command command_id = (Command)packet->data[0];
+    LOG_INFO("Command ID Received: %i", command_id);
+
+    switch (command_id)
+    {
+        /* Payload Commands */
+        case TAKE_PHOTO:
+        case DOWNLOAD_PHOTO:
+        case TAKE_AND_SEND_PHOTO:
+        {
+            PAYLOAD_COMMAND_DATA task;
+            strncpy(task.serialized_command, packet->data + 1,
+                    sizeof(task.serialized_command) - 1);
+            task.serialized_command[sizeof(task.serialized_command) - 1] = '\0';
+
+            if (command_id == TAKE_PHOTO)
+                queue_try_add(&slate->take_photo_task_data, &task);

```

Comment:
I was imagining TAKE_PHOTO passing a specific string of chars which specify some const string "take_photo,[],{}" as the PAYLOAD_COMMAND_DATA's task.serialized_command?

This currently just passes through whatever is received from the command packet's data field. Could we not consolidate this as the previously suggested single command PAYLOAD_EXEC?